spec/features/brexit_checker/question_results_spec.rb

```diff
@@ -65,6 +71,114 @@ def when_i_visit_the_brexit_checker_flow
+    group_titles_ordered = ["Visiting the EU", "Visiting the UK", "Visiting Ireland"]
+
+    find_all(".brexit-checker-audience-citizen section.brexit-checker-actions__group h3").each_with_index do |group_title, i|
+      expect(group_title.text).to eq(group_titles_ordered[i])
+    end
+  end
+
+  def and_all_business_actions_links_should_have_ecommerce_tracking
+    find_all('.brexit-checker-business-actions .brexit-checker__action a.govuk-link').each do |link|
+      expect(link['data-module']).to eq("track-click")
+    end
+  end
+
+  def and_business_results_audience_analyitics_tracking_should_be_present
+    and_business_actions_section_has_ecommerce_tracking
+    and_business_actions_section_has_data_search_query_attributes
+    and_all_business_actions_links_should_have_ecommerce_tracking
+  end
+
+  def and_business_actions_section_has_ecommerce_tracking
+    section = find('.brexit-checker-business-actions', visible: false)
+    expect(section['data-analytics-ecommerce']).to eq("")
+    expect(section['data-ecommerce-start-index']).to eq("1")
+    expect(section['data-list-title']).to eq("Brexit checker results: Your business or organisation")
+  end
+
+  def all_business_action_links_have_ecommerce_tracking(link, action_index, link_type)
+    expect(link['data-module']).to eq("track-click")
+    expect(link['data-track-category']).to eq("brexit-checker-results")
+    expect(link['data-ecommerce-path']).to eq(link['href'].sub("https://www.gov.uk",""))
+    expect(link['data-ecommerce-row']).to eq("")
+    expect(link['data-track-label']).to eq(link['href'])
+    action_string = link['data-track-action'].split(" - ")
+    expect(action_string[0]).to eql(I18n.t!("brexit_checker.results.audiences.business.heading"))
+    expect(action_string[1]).to eql(I18n.t!('brexit_checker.results.audiences.business.heading'))
+    expect(action_string[2]).to eql("1.#{action_index + 1}")
+    expect(action_string[3]).to eql(link_type)
+  end
+
+  def all_citizen_action_links_have_ecommerce_tracking(link, group_index, group_title, action_index, link_type)
+    expect(link['data-module']).to eq("track-click")
+    expect(link['data-track-category']).to eq("brexit-checker-results")
+    expect(link['data-ecommerce-path']).to eq(link['href'].sub("https://www.gov.uk",""))
+    expect(link['data-ecommerce-row']).to eq("")
+    expect(link['data-track-label']).to eq(link['href'])
+    action_string = link['data-track-action'].split(" - ")
+    expect(action_string[0]).to eql(I18n.t!("brexit_checker.results.audiences.citizen.heading"))
+    expect(action_string[1]).to eql(group_title)
+    expect(action_string[2]).to eql("#{group_index}.#{action_index + 1}")
+    expect(action_string[3]).to eql(link_type)
+  end
+
+  def and_all_business_actions_links_should_have_ecommerce_tracking
+    find_all('brexit-checker-business-actions .brexit-checker__action .brexit-checker__action_title a.govuk-link').each_with_index do |link, i|
```

Comment:
I'm not sure how this is working  - doesn't this need a class selector?

```suggestion
    find_all('.brexit-checker-business-actions .brexit-checker__action .brexit-checker__action_title a.govuk-link').each_with_index do |link, i|
```

Reply:
I don't think we need a selector this long - can we not simplify to: `.brexit-checker__action_title .govuk-link`